### DISCUSSION TITLE
What are the steps to install prisma on an offline machine

### DISCUSSION THREAD

**Discussion by SirMatGuy** [prisma/prisma#17536] [Q&A]
Hello everyone!

I need your help to install prisma on an offline machine ( only file transfer)

Basically i already have all the npm packages but when i run
`npx prisma generate`
it fails due to it trying to get the query engine from the internet
So i need to figure out how to get the windows and debian versions of the query engine from an online windows machine 
to my offline windows machine that later i need to create a docker image from

Thank you so much for any help!!

**Comment by nurul3101** [comment]
Hello @SirMatGuy 👋 

Are you copying `node_modules` as well when you are transferring the files to an offline machine?

If you are copying over the node_modules, then by default PrismaClient is generated inside node_modules, so you would not need to invoke `npx prisma generate` on the offline machine.

**Reply by SirMatGuy** [reply]
Hey @nurul3101 !
First thank you for taking your time and helping, really appreciate it
So i guessed i forgot to add, that i have an artifact server, from where i can do
`npm install <package>` for example
After a bit of digging i found that i need to download the engines separately,
The problem is now i want to find darwin and windows engines for prisma, but i’m not sure of the urls i need…

**Reply by nurul3101** [reply]
You are probably looking for URL which are in this format: https://github.com/prisma/prisma/discussions/14873#discussioncomment-4206246

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
